### PR TITLE
Add warning for cache tags limits

### PR DESCRIPTION
**Pull request by ijjk**
We currently silently ignore tags when they exceed the limits but we should warn for them as well so users are aware the limit is hit.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 0bb663a2487c66bacfd272411f47afa26c031853

`pnpm test test/integration/create-next-app/package-manager.test.ts` 
-  > should use Yarn as the package manager on supplying --use-yarn
-  > should infer yarn as the package manager

<details>
<summary>Expand output</summary>

  ● should use Yarn as the package manager on supplying --use-yarn

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:123:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:103:3)

  ● should infer yarn as the package manager

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:433:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:405:3)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| buildDuration | 17.1s | 17.5s | ⚠️ +423ms |
| buildDurationCached | 11.4s | 10.2s | N/A |
| nodeModulesSize | 175 MB | 175 MB | ⚠️ +24.1 kB |
| nextStartRea..uration (ms) | 676ms | 713ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| 199-HASH.js gzip | 30 kB | 30 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.2 kB | 53.2 kB | ✓ |
| 494.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.5 kB | 45.5 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 251 B | N/A |
| main-HASH.js gzip | 33.1 kB | 33.1 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 98.9 kB | 98.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| _app-HASH.js gzip | 205 B | 205 B | ✓ |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 505 B | 507 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.59 kB | 2.59 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 259 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.38 kB | 4.38 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 318 B | 318 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 319 B | 320 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 885 B | 885 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 484 B | ✓ |
| Overall change | 484 B | 484 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| index.html gzip | 529 B | 528 B | N/A |
| link.html gzip | 541 B | 543 B | N/A |
| withRouter.html gzip | 525 B | 524 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| edge-ssr.js gzip | 96.1 kB | 96.1 kB | N/A |
| page.js gzip | 140 kB | 140 kB | ⚠️ +108 B |
| Overall change | 140 kB | 140 kB | ⚠️ +108 B |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/tags-warning | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 627 B | N/A |
| middleware-r..fest.js gzip | 148 B | 151 B | N/A |
| middleware.js gzip | 23 kB | 23 kB | ✓ |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 25 kB | 25 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 0bb663a2487c66bacfd272411f47afa26c031853

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 0e6c8030bc45061128ce867962f7ea4ac0c6d8c8

`pnpm test-dev test/development/basic/next-rs-api.test.ts` 
- next.rs api > should have working HMR on client-side change on a app page 1

<details>
<summary>Expand output</summary>

  ● next.rs api › should have working HMR on client-side change on a app page 1

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      551 |             ).toEqual([expect.stringContaining(expectedUpdate)])
      552 |           }
    > 553 |           expect(foundServerSideChange).toBe(expectedServerSideChange)
          |                                         ^
      554 |           ok = true
      555 |         } finally {
      556 |           try {

      at Object.toBe (development/basic/next-rs-api.test.ts:553:41)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/server/lib/patch-fetch.ts [hunk 1]
```diff
@@ -98,7 +104,19 @@ export function addImplicitTags(staticGenerationStore: StaticGenerationStore) {
...
+      filteredTags.push(tag)
+    } else {
+      newTags.push(tag)
+    }
+  }
+
+  if (filteredTags.length > 0) {
+    console.warn(
+      `Tags exceeded limit of 64 items or 1024 max length per item on ${urlPathname},  filtered tags: `,
+      filteredTags.join(',')
```
Comment (lubakravche) [inline packages/next/src/server/lib/patch-fetch.ts:118]: maybe trim this?

**Pull request closed without merging**